sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1821,6 +1821,47 @@ -(void) testIncrementalPullWithInsert
     XCTAssertEqual(client.syncContext.pendingOperationsCount, 1);
 }
 
+// Tests incremental pull with interleaved local operations is Id case insensitive
+-(void)testIncrementalPullIsIdCaseInsensitive
+{
+    // Response of a pull operation
+    NSString *firstPullData = @"[{\"id\": \"one\", \"text\":\"first item\", \"__updatedAt\":\"1999-12-03T15:44:29.0Z\"},{\"id\": \"two\", \"text\":\"second item\", \"__updatedAt\":\"1999-12-03T15:44:28.0Z\"}]";
+
+    // Setup filter to get the pull response
+    MSMultiRequestTestFilter *filter = [MSMultiRequestTestFilter testFilterWithStatusCodes:@[@200] data:@[firstPullData] appendEmptyRequest:YES];
+
+    // Initialization
+    MSClient *filteredClient = [client clientWithFilter:filter];
+    MSSyncTable *todoTable = [filteredClient syncTableWithName:@"TodoItem"];
+    MSQuery *query = [[MSQuery alloc] initWithSyncTable:todoTable];
+
+    // Perform an insert operation in the middle of a pull operation
+    ((MSTestFilter *)filter.testFilters[0]).onInspectRequest = ^(NSURLRequest *request) {
+
+        // Insert an item with Id differing from that of the pulled item only in case
+        NSDictionary *item = @{ @"id": @"ONE", @"name":@"fifth item" };
+        MSTestWaiter *insertWaiter = [MSTestWaiter new];
+    
+        [todoTable insert:item completion:^(NSDictionary *item, NSError *error) {
```

Comment:
this isn't going to simulate the standard pull merge into an id logic. I thought the client will detect the record changed and not even try to upsert now.